src/trajectory.jl

```diff
@@ -25,7 +25,7 @@ end
 "Returns the statistics for transition `t`."
 stat(t::Transition) = t.stat
 
-abstract type AbstractProposal end
+abstract type AbstractMCMCKernel end
```

Comment:
The one annoying bit with this is that we're really moving towards reimplementing parts of AbstractMCMC.jl. Ideally we'd just work off that. 

But I guess that's work for another PR? :confused:

Reply:
I guess it's fine for now. 
AbstractMCMC.jl ATM is mostly defining _sampler_ interface (i.e. how `sample` works by abstracting out how parallel stuff should be done, how samples should be collected, etc.) while this _kernel_ interface is for controlling how to manipulate kernels (for now the only thing we will do is making a mixture of a few).
But I agree this kernel interface should also be moved to an upsteam package ideally.

Reply:
I agree, probably something for a later point:)